data/wp/wp-content/mu-plugins/EPFL-SC-infoscience.php

```diff
@@ -38,33 +38,37 @@ function epfl_infoscience_urlExists($url)
     curl_close($handle);
 }
 
-function epfl_infoscience_process_shortcode($attributes, $content = null)
+function epfl_infoscience_process_shortcode( $attributes, $content = null )
 {
-    extract(shortcode_atts(array(
+    $attributes = shortcode_atts( array(
         'url' => ''
-    ), $attributes));
+    ), $attributes);
+
+    // Sanitize parameter
+    $url = sanitize_text_field( $attributes['url'] );
 
     // Check if the result is already in cache
-    $result = wp_cache_get($url, 'epfl_infoscience');
-    if (false === $result){
-        if (strcasecmp(parse_url($url, PHP_URL_HOST), 'infoscience.epfl.ch') == 0 && epfl_infoscience_urlExists($url)) {
+    $result = wp_cache_get( $url, 'epfl_infoscience' );
+    if ( false === $result ){
+        if ( strcasecmp( parse_url( $url, PHP_URL_HOST ), 'infoscience.epfl.ch' ) == 0 && epfl_infoscience_url_exists( $url ) ) {
 
             // Get the content of the cache
-            $page = file_get_contents($url);
+            $page = file_get_contents( $url );
```

Comment:
La fonction `file_get_contents` peut potentiellement être bloquée sur certains serveurs. J'avais quelques surprise avec de par le passé. Après petite recherche, j'avais vu qu'il était plus élégant d'utiliser CURL pour faire les téléchargements de contenu de page. Tu pourrais donc faire comme ceci :

>$curl = curl_init();
curl_setopt($curl, CURLOPT_URL, $url);
      curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
      curl_setopt($curl, CURLOPT_AUTOREFERER, true);
      curl_setopt($curl, CURLOPT_HEADER, false);
      curl_setopt($curl, CURLOPT_USERAGENT, "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:27.0) Gecko/20100101 Firefox/27.0");
      $page = curl_exec($curl);
      curl_close($curl);
      
L'initialisation de CURLOPT_USERAGENT n'est pas obligatoire cependant.

Reply:
D'après cet article https://tommcfarlin.com/alternative-to-file_get_contents/ et 1er commentaire de dylan, dans le cas de WP, il est apparement plus judicieux d'utiliser 
 
`$response = wp_remote_get( ‘https://api.github.com/users/blobaugh’ );
$body = wp_remote_retrieve_body( $response );`